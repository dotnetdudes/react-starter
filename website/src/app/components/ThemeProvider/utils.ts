import { ThemeType } from './slice/types';

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined;

export function saveTheme(theme: ThemeType) {
  window.localStorage && localStorage.setItem('selectedTheme', theme);
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeType | null {
  return window.localStorage
    ? (localStorage.getItem('selectedTheme') as ThemeType) || null
    : null;
}
