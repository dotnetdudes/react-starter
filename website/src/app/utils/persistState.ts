// @flow
// https://egghead.io/lessons/javascript-redux-persisting-the-state-to-the-local-storage
// https://medium.com/@jrcreencia/persisting-redux-state-to-local-storage-f81eb0b90e7e
import type { RootState } from 'types';

export const loadState = () => {
  try {
    const serializedState = sessionStorage.getItem('stateSramReg');
    if (serializedState === null) {
      return undefined;
    }
    // $FlowIgnore
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

export const saveState = (state: RootState) => {
  try {
    const serializedState = JSON.stringify(state);
    sessionStorage.setItem('stateSramReg', serializedState);
  } catch (err) {
    // die
  }
};
