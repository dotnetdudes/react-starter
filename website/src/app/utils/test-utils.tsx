import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureAppStore } from 'store/configureStore';
import { ThemeProvider, createTheme } from '@mui/material/styles';

const store = configureAppStore();

const MockTheme = ({ children }: any) => {
  const theme = createTheme({});
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <Provider store={store}>
      <MockTheme>
        <BrowserRouter>{children}</BrowserRouter>
      </MockTheme>
    </Provider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
