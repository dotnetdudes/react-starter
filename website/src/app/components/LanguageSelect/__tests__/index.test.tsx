import { LanguageSelect } from '..';
import { render } from 'app/utils/test-utils';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<LanguageSelect  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<LanguageSelect />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
