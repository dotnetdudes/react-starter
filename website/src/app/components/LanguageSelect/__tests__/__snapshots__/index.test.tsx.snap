// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelect  /> should match snapshot 1`] = `
<div
  class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink="true"
    id="language-select-label"
  >
    age
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="language-select-label language-select"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="language-select"
      role="button"
      tabindex="0"
    >
      english
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      tabindex="-1"
      value="en"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-14lo706"
      >
        <span>
          age
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
